src/node.cc

```diff
@@ -840,6 +842,26 @@ void SetupNextTick(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(ret);
 }
 
+void WeakCallback(
+    const v8::WeakCallbackInfo<v8::Persistent<v8::Promise,
+      node::ResetInDestructorPersistentTraits<v8::Promise> > >& data) {
+  Environment* env = Environment::GetCurrent(data.GetIsolate());
+  Persistent<Promise>* p = data.GetParameter();
+  Local<Promise> promise = p->Get(env->isolate());
+  Local<Value> err = promise->Result();
+  Persistent<Value>* e = new Persistent<Value>(env->isolate(), err);
+
+  env->SetImmediate([](Environment* env, void* data) {
+    Local<Value>* e = static_cast< Local<Value>* >(data);
```

Comment:
Regarding those failures in CI … this was a `Persistent<Value>`, so you need to cast back to that type as well. Also, you need a `HandleScope` here, because you are creating new handles inside the function as well.

(And style nit: No spaces after/before `<>`)